File: spec2code/java-spring/src/main/java/de/codecentric/javaspring/persistence/Book.java
package de.codecentric.javaspring.persistence;

import java.util.Set;
import java.util.UUID;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Data;
import lombok.Setter;

@Data
@Builder
@Setter(AccessLevel.PRIVATE)
public class Book implements Literature {
    private final UUID id;
    private final String title;
    private final Set<UUID> authors;
    private final String recap;
    private final Genre genre;
    private final String isbn;
    private final int pages;
}

File: spec2code/kotlin-spring/src/main/kotlin/de/codecentric/kotlinspring/persistence/Book.kt
package de.codecentric.kotlinspring.persistence

import java.util.UUID

data class Book(
    override val id: UUID,
    override val title: String,
    override val authors: Set<UUID>,
    override val recap: String?,
    override val genre: Genre?,
    val isbn: String,
    val pages: Int,
) : Literature(id, title, authors, recap, genre)
